### ISSUE TITLE
Add support for Kotlin Coroutines

### ISSUE THREAD

**Issue by jamesward** [spring-projects-experimental/spring-native#409]
I have a basic Kotlin coroutine controller method:
```
	@GetMapping("/")
	suspend fun index() = run {
		"aasdf"
	}
```

Works fine in the JVM but fails at runtime when using a native image (`org.springframework.experimental:spring-graalvm-native:0.8.4`):
```
java.lang.IllegalStateException: No primary or single public constructor found for interface kotlin.coroutines.Continuation - and no default constructor found either
	at org.springframework.beans.BeanUtils.getResolvableConstructor(BeanUtils.java:250) ~[na:na]
```

Without the `suspend` it works as expected:
```
	@GetMapping("/")
	fun index(): String {
		return "asdf"
	}
```

**Comment by jamesward** [comment]
Here is a project that reproduces the error (`graalvm` branch):
https://github.com/jamesward/springboot-kotlin-reactive-demo/tree/graalvm

**Comment by sdeleuze** [comment]
If have switch `KotlinDetector` initialization to build time and order to fix that, then added `Continuation` related reflection entries but that seems not sufficient, see https://github.com/spring-projects-experimental/spring-graalvm-native/tree/webflux-kotlin.

I suspect `kotlin-reflect` to be too reflection based and big to be really a good fit for native, so in parrallel on working on that I have created https://github.com/spring-projects/spring-framework/issues/21546 which would be an ambitious but IMO promising way to solve that.

**Comment by sdeleuze** [comment]
For the time being I have [documented that Coroutines are not supported yet](https://github.com/spring-projects-experimental/spring-graalvm-native/commit/219bbb015d3d8abe3a1b00cc15d637245658ec16).

**Comment by sdeleuze** [comment]
We should be able to use Coroutines with proper documentation (add required entries with `@NativeHint`) until [KT-44594](https://youtrack.jetbrains.com/issue/KT-44594) fix is available in Kotlin 1.5-M2. To be check with a modified version of `webmvc-kotlin` that uses WebFlux and Coroutines.
